src/core.js

```diff
@@ -1540,6 +1540,18 @@ const _makeWindow = (options = {}, parent = null, top = null) => {
     }
   });
 
+  // Block polyfills from overwriting Promise object.
+  const originalWindowPromise = window.Promise;
+  Object.defineProperty(window, 'Promise', {
```

Comment:
Is `writable`/`enumerable` more appropritate? Probably makes more sense than a function call trap.

```
$ node
x = {};
Object.defineProperty(x, 'lol', { value: 'zol', writable: false, enumerable: true, });
x.lol = 'woot';
x.lol;
```